### DISCUSSION TITLE
[Feature] Random animations for Add-ons and Stages

### DISCUSSION THREAD

**Discussion by MUGEN-Slayer** [ikemen-engine/Ikemen-GO#846] [Feature Request]
It would work similarly to "Begin Action", but would use blocks. With "start" and "end" to define each block. That way, every time you access the Menu or Options for example, the engine would choose randomly a block to start playing the animation. Once the animation started playing, it would read the next block in a loop, in a similar way to Begin Action reading the next lines.
Being also possible in lifebars and stages would be great.

    [TitleBG 1]
    type = random
    randomizerno = 1
    mask = 1
    start = 0,0
     
    [Randomizer 1]
    block1.start
    50,0, 0,0, 3,,AS96D160
    50,0, 0,0, 3,,AS128D128
    50,0, 0,0, 3,,AS160D96
    50,0, 0,0, 3,,AS192D64
    50,0, 0,0, 50
    50,0, 0,0, 3,,AS192D64
    50,0, 0,0, 3,,AS192D64
    50,0, 0,0, 3,,AS128D128
    50,0, 0,0, 3,,AS96D160
    -1,0, 0,0, 30
    block1.end
    block2.start
    50,1, 0,0, 3,,AS96D160
    50,1, 0,0, 3,,AS128D128
    50,1, 0,0, 3,,AS160D96
    50,1, 0,0, 1,,AS192D64
    50,1, 0,0, 50
    50,1, 0,0, 3,,AS192D64
    50,1, 0,0, 3,,AS192D64
    50,1, 0,0, 3,,AS128D128
    50,1, 0,0, 3,,AS96D160
    -1,0, 0,0, 30
    block2.end
    block3.start
    50,2, 0,0, 3,,AS96D160
    50,2, 0,0, 3,,AS128D128
    50,2, 0,0, 3,,AS160D96
    50,2, 0,0, 1,,AS192D64
    50,2, 0,0, 50
    50,2, 0,0, 3,,AS192D64
    50,2, 0,0, 3,,AS192D64
    50,2, 0,0, 3,,AS128D128
    50,2, 0,0, 3,,AS96D160
    -1,0, 0,0, 30
    block3.end

**Comment by LasombraDemon** [comment]
As simplified as this version is, it's already quite doable with AttachedChar.

**Reply by MUGEN-Slayer** [reply]
The idea here would be to have something similar to the current animations (begin actions), without having to literally create a (attached) character. Also, something that would work in screenpacks, lifebars and other add-ons.

**Comment by potsmugen** [comment]
I think a simpler implementation would be to somehow specify more than one animation and the engine would pick randomly.

**Reply by LasombraDemon** [reply]
Then again, stages could use a whole randomrange-like function. Not just for animations, but for object positions and certain timings.

You can do all these with AttachedChar, but it's totally overkill to make one for such a simple feature.

**Reply by MUGEN-Slayer** [reply]
That would work too.
